packages/knip/src/util/file-entry-cache.ts

```diff
@@ -38,7 +40,19 @@ export class FileEntryCache<T> {
   normalizedEntries = new Map<string, FileDescriptor<T>>();
 
   constructor(cacheId: string, _path: string) {
-    this.filePath = isAbsolute(_path) ? resolve(_path, cacheId) : resolve(cwd, _path, cacheId);
+    /**
+     * analyzes process for posix vs. win32 environment distinction
+     * applies correct path transformation based on the current environment
+     * and applies it to the cache location passed in from the cli args
+     */
+    const isWindows = process.platform === 'win32';
+    const { directory: directoryFromCliArg } = parsedArgValues;
+    const currentCwd = directoryFromCliArg || cwd;
+    const currentPath = isWindows ? path.win32.resolve(currentCwd): path.posix.resolve(currentCwd);
+    const pathResolver =  isWindows ? path.win32.join : resolve;
+
+    this.filePath = isAbsolute(currentPath) ? pathResolver(currentPath, cacheId) : pathResolver(currentPath, _path, cacheId);
```

Comment:
- `cwd` should already `[be --directory](https://github.com/webpro-nl/knip/blob/3ea698db8386e994c8df5580c1aea65d21ca39f2/packages/knip/src/util/path.ts#L19) so i don't think we need to read `directory` here from `parsedArgValues` again
- we can use `path.resolve` which should internally do the platform switch, no?
- same for `path.join.

So, thinking out loud, would only this do the same?

```ts
    const currentPath = path.resolve(cwd);
    this.filePath = isAbsolute(currentPath) ? path.join(currentPath, cacheId) : path.join(currentPath, _path, cacheId);
```

Reply:
> * `cwd` should already [be --directory](https://github.com/webpro-nl/knip/blob/3ea698db8386e994c8df5580c1aea65d21ca39f2/packages/knip/src/util/path.ts#L19) so i don't think we need to read `directory` here from `parsedArgValues` again
> * we can use `path.resolve` which should internally do the platform switch, no?
> * same for `path.join`
> 
> So, thinking out loud, would only this do the same?
> 
> ```ts
>     const currentPath = path.resolve(cwd);
>     this.filePath = isAbsolute(currentPath) ? path.join(currentPath, cacheId) : path.join(currentPath, _path, cacheId);
> ```

I will check on those things. We could probably simplify the approach to resolving the path - I'd like to verify that we only need the generic path resolver and not win32 specific one. Regarding directory - I meant to use `cache-location` instead. I will fix this in my next commit.

Reply:
Thanks! I think `cache-location` is the `_path` we're receiving as the second argument.

I'm seeing now I made a mistake, here's the correction (did not actually test):

```ts
const currentPath = path.resolve(cwd);
this.filePath = isAbsolute(_path) ? path.join(_path, cacheId) : path.join(currentPath, _path, cacheId);
```